Source/WebKit/Platform/IPC/Decoder.cpp

```diff
@@ -27,58 +27,52 @@
 #include "Decoder.h"
 
 #include "ArgumentCoders.h"
-#include "DataReference.h"
 #include "Logging.h"
 #include "MessageFlags.h"
 #include <stdio.h>
 #include <wtf/StdLibExtras.h>
 
 namespace IPC {
 
-static const uint8_t* copyBuffer(const uint8_t* buffer, size_t bufferSize)
+static const uint8_t* copyBuffer(DataReference buffer)
 {
     uint8_t* bufferCopy;
+    const size_t bufferSize = buffer.size_bytes();
     if (!tryFastMalloc(bufferSize).getValue(bufferCopy)) {
         RELEASE_LOG_FAULT(IPC, "Decoder::copyBuffer: tryFastMalloc(%lu) failed", bufferSize);
         return nullptr;
     }
 
-    memcpy(bufferCopy, buffer, bufferSize);
+    memcpy(bufferCopy, buffer.data(), bufferSize);
     return bufferCopy;
 }
 
-std::unique_ptr<Decoder> Decoder::create(const uint8_t* buffer, size_t bufferSize, Vector<Attachment>&& attachments)
+std::unique_ptr<Decoder> Decoder::create(DataReference buffer, Vector<Attachment>&& attachments)
```

Comment:
Not a regression from this patch but it seems wrong to use a const pointer (and now a `span<const uint8_t>`) to represent memory we need to free.

Reply:
Well, sometimes we free it (if `BufferDeallocator` is set) and sometimes we don't free it (if `BufferDeallocator` is `nullptr`).

Are you suggesting changing `DataReference` to be `std::span<uint8_t>` everywhere, or just change one `Decoder` constructor and one `Decoder::create()` method to use `std::span<uint8_t>` in place of `DataReference`?

Or do we need a `MutableDataReference` that's `std::span<uint8_t>` to differentiate `DataReference` that's `std::span<const uint8_t>`?  (I tried this, and it got messy quickly, so I think this should probably done in a separate patch.)

Reply:
See above.